mlir/lib/Dialect/Affine/IR/AffineOps.cpp

```diff
@@ -1902,6 +1902,10 @@ LogicalResult AffineForOp::verifyRegions() {
     if (failed(verifyDimAndSymbolIdentifiers(*this, getUpperBoundOperands(),
                                              getUpperBoundMap().getNumDims())))
       return failure();
+  if (getLowerBoundMap().getNumResults() < 1)
+    return emitOpError("expected lower bound map to have at least one result");
+  if (getUpperBoundMap().getNumResults() < 1)
+    return emitOpError("expected upper bound map to have at least one result");
```

Comment:
`affine.parallel` op's verifier needs this same fix. Fine to merge this one first.

Reply:
I've taken care of fixing it for affine.parallel here: https://github.com/llvm/llvm-project/pull/127611